Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -306,46 +342,106 @@ void Device::rasterize_triangle(const Triangle& triangle)
 
             int coverage_bits = maskbits(quad.mask);
 
+            // Stencil testing
+            u8* stencil_ptrs[4];
+            i32x4 stencil_value;
+            if (m_options.enable_stencil_test) {
+                stencil_ptrs[0] = coverage_bits & 1 ? &m_stencil_buffer->scanline(by)[bx] : nullptr;
+                stencil_ptrs[1] = coverage_bits & 2 ? &m_stencil_buffer->scanline(by)[bx + 1] : nullptr;
+                stencil_ptrs[2] = coverage_bits & 4 ? &m_stencil_buffer->scanline(by + 1)[bx] : nullptr;
+                stencil_ptrs[3] = coverage_bits & 8 ? &m_stencil_buffer->scanline(by + 1)[bx + 1] : nullptr;
+
+                stencil_value = load4_masked(stencil_ptrs[0], stencil_ptrs[1], stencil_ptrs[2], stencil_ptrs[3], quad.mask);
+                stencil_value &= stencil_configuration.test_mask;
+
+                i32x4 stencil_test_passed;
+                switch (stencil_configuration.test_function) {
+                case StencilTestFunction::Always:
+                    stencil_test_passed = expand4(~0);
```

Comment:
Why not `expand4(0xffffffffu)`?

Reply:
`~0` is synonymous to 'set all bits to 1' to me, and prevents mistyping this like `0xfffffffu` for example. But if there is some other disadvantage I'm missing, I'm all ears!

Reply:
Nope, completely stylistic! `~0` is completely valid too, though I'm usually one to miss the tilde.